language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
Unfortunately (by design) all variables in a class are externally available and thus part of the class' API. Even more unfortunate is that the set of variables can be conditional.
This does not alter the old design decisions though. Instead, this is intended to help in situations where something private/implementation specific should be exposed as a set of documented and supported variables in a modules "API classes". You can do the same now by just typing a lot. The interesting thing you can now do (which was horrible to do in the past), is to assign variables from a selection of class based on a  variable - for example [$a, $x] = Class[$the_inner_class_used], which allows an advanced module that configures itself to easily choose which variables to expose - result is a lot clearer than lots of lines of variable assignment intermixed with a bit of if-then-else logic.